cmd/containerd-shim-runhcs-v1/rootfs.go

```diff
@@ -56,17 +56,10 @@ func parseLegacyRootfsMount(m *types.Mount) (string, []string, error) {
 	//
 	// The OCI spec expects:
 	//   layerN, layerN-1, ..., layer0, scratch
-	var parentLayerPaths []string
-	for _, option := range m.Options {
-		if strings.HasPrefix(option, mount.ParentLayerPathsFlag) {
-			err := json.Unmarshal([]byte(option[len(mount.ParentLayerPathsFlag):]), &parentLayerPaths)
-			if err != nil {
-				return "", nil, fmt.Errorf("unmarshal parent layer paths from mount: %v: %w", err, errdefs.ErrFailedPrecondition)
-			}
-			// Would perhaps be worthwhile to check for unrecognized options and return an error,
-			// but since this is a legacy layer mount we don't do that to avoid breaking anyone.
-			break
-		}
+	containerdMount := mount.Mount{Type: m.Type, Source: m.Source, Target: m.Target, Options: m.Options}
+	parentLayerPaths, err := containerdMount.GetParentPaths()
+	if err != nil {
+		return "", nil, fmt.Errorf("unmarshal parent layer paths from mount: %v: %w", err, errdefs.ErrFailedPrecondition)
```

Comment:
Nit: This error message is essentially the same as the one in containerd/mount. Consider changing to something like below:

```suggestion
		return "", nil, fmt.Errorf("failed to get mount's parent layer paths: %v: %w", err, errdefs.ErrFailedPrecondition)
```